File: languages/go/entity/identity/date_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DateAttributeDescriptor represents a descriptor for a date attribute, including its display name, ID, description, and required status.
type DateAttributeDescriptor struct {
	AttributeDescriptor
}

// NewDateAttributeDescriptor creates a new DateAttributeDescriptor instance with the provided details.
func NewDateAttributeDescriptor(name string, id string, description string, required bool) DateAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return DateAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewDateAttributeDescriptorFromProto(s *proto.DateAttributeDefinition) DateAttributeDescriptor {
	if s == nil {
		return DateAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return DateAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s DateAttributeDescriptor) ToProto() *proto.DateAttributeDefinition {
	return &proto.DateAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/date_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for a date attribute, including its display name, ID, description, and required status.
 */
export class DateAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.DateAttributeDefinition {
    return identityEntitiesProto.DateAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.DateAttributeDefinition
  ): DateAttributeDescriptor {
    return new DateAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
